File: Data Structures & Algorithms/intersection-of-two-arrays/submission-1.go
func intersection(nums1 []int, nums2 []int) []int {
    seen := make(map[int]struct{},len(nums1))
	result := make([]int,0)
	for _, val := range nums1 {
		seen[val] = struct{}{}
	}

	for _, val := range nums2 {
		if _,ok := seen[val]; ok  {
			result = append(result,val)
			delete(seen,val)
		}
	}
	return result
}

File: Data Structures & Algorithms/find-the-difference-of-two-arrays/submission-1.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        set1 = set(nums1)
        set2 = set(nums2)
        result1 = []
        result2 = []

        for i in range(len(nums1)):
            if nums1[i] not in set2 and nums1[i] not in result1:
                result1.append(nums1[i])
            
        for i in range(len(nums2)):
            if nums2[i] not in set1 and nums2[i] not in result2:
                result2.append(nums2[i])

        result = []
        result.append(result1)
        result.append(result2)
        return result